### PR TITLE
fix(import): make dropdown mandatory

### PR DESCRIPTION
**Pull request by abhinandan13jan**
## Fixes 

https://issues.redhat.com/browse/KFLUXBUGS-1745

## Description

Makes selection of git-provider annotation mandatory
Adds https:// to the git-url annotation

## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

Screenshot 2024-10-18 at 4 56 25 PM
Screenshot 2024-10-18 at 4 56 39 PM
Screenshot 2024-10-18 at 4 58 16 PM

## Browser conformance: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: src/components/ImportForm/ComponentSection/SourceSection.tsx [hunk 1]
```diff
@@ -39,7 +39,7 @@ export const SourceSection = () => {
         let name: string;
         try {
           parsed = GitUrlParse(event.target?.value ?? '');
-          setFieldValue('gitURLAnnotation', parsed?.resource);
+          setFieldValue('gitURLAnnotation', `https://${parsed?.resource}`);
```
Comment (MartinBasti) [inline src/components/ImportForm/ComponentSection/SourceSection.tsx:42]: Maybe avoid hardcoding
```suggestion
          setFieldValue('gitURLAnnotation', `${parsed?.protocol}://${parsed?.resource}`);
```

**Comment by testcara** [comment]
Hi, I created one related PR https://github.com/openshift/hac-dev/pull/1015 which is based on yours.
I went through the [KFLUXBUGS-1786](https://issues.redhat.com/browse/KFLUXBUGS-1786) & [KFLUXBUGS-1786](https://issues.redhat.com/browse/KFLUXBUGS-1785), it seems users expects when inputting 'https://gitlab.cee.redhat.com/test.git', the git provider should be gitlab and https://gitlab.cee.redhat.com.

The PR fixed the https, [!1015](https://github.com/openshift/hac-dev/pull/1015) ensure it's gitlab auto-filled for that case .
Please also help to review [!1015](https://github.com/openshift/hac-dev/pull/1015) to confirm it is useful or not.
Thank you so much.

**Comment by testcara** [comment]
BTW, if we just support github & gitlab, do we need to do some code cleanup for the original other type git provider like 'bitbucket.org'?

**Comment by testcara** [comment]
FYI. https://github.com/openshift/hac-dev/pull/1015 has been closed directly. Thank you for your help and suggestion.

**Review by sahil143 — APPROVED** [review]
/approve

**Pull request merged**